fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryTest.java

```diff
@@ -233,4 +241,62 @@ private void assertMetricCountAtMost(StoreTimer.Event metric, int maximumValue)
         assertThat(String.format("Metric %s should be called at most %d times", metric, maximumValue),
                 timer.getCount(metric), lessThanOrEqualTo(maximumValue));
     }
+
+
+    @Test
+    void testFileLock() throws IOException {
+        String lockName = "file.lock";
+        final Lock lock1 = directory.obtainLock(lockName);
+        lock1.ensureValid();
+        RecordCoreException e = assertThrows(RecordCoreException.class, () -> directory.obtainLock(lockName));
+        assertTrue(e.getMessage().contains("found old lock"));
+        lock1.ensureValid();
+        lock1.close();
+        assertThrows(AlreadyClosedException.class, lock1::ensureValid);
+        final Lock lock2 = directory.obtainLock(lockName);
+        lock2.ensureValid();
+        e = assertThrows(RecordCoreException.class, () -> directory.obtainLock(lockName));
+        assertTrue(e.getMessage().contains("found old lock"));
+        lock2.ensureValid();
+        lock2.close();
+    }
+
+    @Test
+    void testAgilityContext() throws ExecutionException, InterruptedException {
+        try (FDBRecordContext context = fdb.openContext()) {
+            final AgilityContext agilityContext = AgilityContext.factory(context, true);
+            agilityContextTestSingleThread(0, agilityContext);
+            agilityContextTestSingleThread(1, agilityContext);
+            agilityContextTestSingleThread(1, agilityContext);
+
+            for (int loop = 0; loop < 20; loop++) {
+                IntStream.rangeClosed(0, 17).parallel().forEach(i -> {
+                    try {
+                        agilityContextTestSingleThread(i, agilityContext);
+                    } catch (ExecutionException | InterruptedException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+            }
+        }
+    }
+
+    private void agilityContextTestSingleThread(int i, AgilityContext agilityContext) throws ExecutionException, InterruptedException {
+        byte[] key = Tuple.from(500, i).pack();
+        byte[] val = Tuple.from(i).pack();
+        try {
+            agilityContext.set(key, val);
+            if (random.nextInt(10) > 7) { // keep some un-flushed values
+                final byte[] bytes = agilityContext.get(key).get();
```

Comment:
`get`/`set` are both very fast, would it improve the coverage of concurrency by using `apply`/`accept` that do a bit more? Or perhaps having it fast is better? Maybe both would be worthwhile.

Reply:
Implemented in the last test.